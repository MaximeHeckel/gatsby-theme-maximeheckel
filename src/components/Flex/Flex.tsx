import styled from '@emotion/styled';

const Flex = styled.div<{
  justifyContent?: string;
  direction?: string;
  wrap?: string;
}>`
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  flex-wrap: ${(p) => p.wrap || 'nowrap'};
  flex-direction: ${(p) => p.direction || 'row'};
  justify-content: ${(p) => p.justifyContent || 'flex-start'};

  > * {
    margin-right: 5px;
    margin-bottom: 0px;
  }
`;

export { Flex };
