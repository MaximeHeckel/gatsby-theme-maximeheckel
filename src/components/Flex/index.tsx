import styled from '../../utils/styled';

const Flex = styled.div<{ justifyContent?: string }>`
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  justify-content: ${(p) => p.justifyContent || 'flex-start'};

  * {
    margin-bottom: 0px;
    margin-right: 5px;
  }
`;

export default Flex;
