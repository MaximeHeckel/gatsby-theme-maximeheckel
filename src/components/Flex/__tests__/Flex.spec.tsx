import { cleanup, render } from '@testing-library/react';
import 'jest-dom/extend-expect';
import React from 'react';
import Flex from '../';

describe('Flex', () => {
  beforeEach(cleanup);

  it('renders Flex with justifyContent prop', () => {
    const component = render(<Flex justifyContent="center">Test</Flex>);
    expect(component.baseElement).toMatchSnapshot();
  });

  it('renders Flex with direction prop', () => {
    const component = render(<Flex direction="column">Test</Flex>);
    expect(component.baseElement).toMatchSnapshot();
  });

  it('renders Flex with wrap prop', () => {
    const component = render(<Flex wrap="wrap">Test</Flex>);
    expect(component.baseElement).toMatchSnapshot();
  });
});
