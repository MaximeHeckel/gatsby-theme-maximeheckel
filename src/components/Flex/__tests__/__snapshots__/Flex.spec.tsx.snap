// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex renders Flex with direction prop 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.emotion-0 > * {
  margin-right: 5px;
  margin-bottom: 0px;
}

<body>
  <div>
    <div
      class="emotion-0"
      direction="column"
    >
      Test
    </div>
  </div>
</body>
`;

exports[`Flex renders Flex with justifyContent prop 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-0 > * {
  margin-right: 5px;
  margin-bottom: 0px;
}

<body>
  <div>
    <div
      class="emotion-0"
    >
      Test
    </div>
  </div>
</body>
`;

exports[`Flex renders Flex with wrap prop 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.emotion-0 > * {
  margin-right: 5px;
  margin-bottom: 0px;
}

<body>
  <div>
    <div
      class="emotion-0"
      wrap="wrap"
    >
      Test
    </div>
  </div>
</body>
`;
